ast.go

```diff
@@ -111,43 +161,74 @@ type LiteralExpression struct {
 	Annotation Annotation // Optional: FunctionAnnotation, PrivateUseAnnotation, or ReservedAnnotation
 }
 
+func (le LiteralExpression) String() string {
+	if le.Annotation == nil {
+		return fmt.Sprintf("{ %s }", le.Literal)
+	}
+
+	return fmt.Sprintf("{ %s %s }", le.Literal, le.Annotation)
+}
+
 type VariableExpression struct {
 	Expression
 
 	Annotation Annotation // Optional: FunctionAnnotation, PrivateUseAnnotation, or ReservedAnnotation
 	Variable   Variable
 }
 
+func (ve VariableExpression) String() string {
+	if ve.Annotation == nil {
+		return fmt.Sprintf("{ %s }", ve.Variable)
+	}
+
+	return fmt.Sprintf("{ %s %s }", ve.Variable, ve.Annotation)
+}
+
 type AnnotationExpression struct {
 	Expression
 
 	Annotation Annotation // FunctionAnnotation, PrivateUseAnnotation, or ReservedAnnotation
 }
 
+func (ae AnnotationExpression) String() string { return fmt.Sprintf("{ %s }", ae.Annotation) }
+
 // ---------------------------------Literal------------------------------------
 
 type QuotedLiteral string
 
 func (QuotedLiteral) node()    {}
 func (QuotedLiteral) literal() {}
+func (ql QuotedLiteral) String() string {
+	// quoted-escape = backslash ( backslash / "|" )
+	r := strings.NewReplacer(
+		`\`, `\\`,
+		`|`, `\|`,
+	)
+
+	return fmt.Sprintf("|%s|", r.Replace(string(ql)))
+}
 
 type UnquotedLiteral struct {
 	Literal
 
 	Value Unquoted // NameLiteral or NumberLiteral
 }
 
+func (ul UnquotedLiteral) String() string { return fmt.Sprint(ul.Value) }
+
 type NameLiteral string
 
-func (NameLiteral) node()     {}
-func (NameLiteral) literal()  {}
-func (NameLiteral) unquoted() {}
+func (NameLiteral) node()             {}
+func (NameLiteral) literal()          {}
+func (NameLiteral) unquoted()         {}
+func (nl NameLiteral) String() string { return string(nl) }
 
 type NumberLiteral float64
 
-func (NumberLiteral) node()     {}
-func (NumberLiteral) literal()  {}
-func (NumberLiteral) unquoted() {}
+func (NumberLiteral) node()             {}
+func (NumberLiteral) literal()          {}
+func (NumberLiteral) unquoted()         {}
+func (nl NumberLiteral) String() string { return fmt.Sprint(float64(nl)) }
```

Comment:
is type conversion to float64 necessary?

Reply:
```go
fmt.Sprint(nl)
```

This one will go into infinite recursion :)